lib/migrate.js

```diff
@@ -0,0 +1,444 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-connector-couchdb2
+// This file is licensed under the Apache License 2.0.
+// License text available at https://opensource.org/licenses/Apache-2.0
+
+'use strict';
+
+var async = require('async');
+var assert = require('assert');
+var g = require('strong-globalize')();
+var util = require('util');
+var _ = require('lodash');
+
+module.exports = mixinMigrate;
+
+function mixinMigrate(CouchDB) {
+  var debug = require('debug')('loopback:connector:couchdb:migrate');
+
+/**
+ * Perform automigrate for the given models.
+ * - destroys model data if the model exists
+ * - delete existing indexes in database
+ * - create new indexes according to model definition
+ * 
+ * @param {String|String[]} [models] A model name or an array of model names.
+ * If not present, apply to all models
+ * @callback {Function} cb The callback function
+ */
+CouchDB.prototype.automigrate = function(models, cb) {
+  debug('CouchDB.prototype.automigrate models %j', models);
+  var self = this;
+  var existingModels = models;
```

Comment:
Can we check if `models` is actually a list of models or a model here, otherwise select all models? We might run into an issue when we run the `async.eachSeries` call down the line